### PR TITLE
Initial stab at an API reference

### PR DESCRIPTION
**Pull request by ebassi**
No unused symbols, but less than 10% actual coverage.

### CODE REVIEW

File: docs/gthree-docs.xml [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  <part id="base-types">
+    <title>Types</title>
+
+    <chapter>
+      <title>Objects</title>
+      <xi:include href="xml/gthreeobject.xml" />
+      <xi:include href="xml/gthreecamera.xml" />
+      <xi:include href="xml/gthreelight.xml" />
+      <xi:include href="xml/gthreeresource.xml" />
+      <xi:include href="xml/gthreematerial.xml" />
```
Comment (alexlarsson) [inline docs/gthree-docs.xml]: Shouldn't materials be under materials?

Comment (alexlarsson) [reply docs/gthree-docs.xml]: Unclear.. camera is not under cameras...

File: docs/resources.xml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      <para>
+        All objects that may hold GL resources are of the type
+        GthreeResource<footnote><para>Actually programs are not, but all
+        programs are owned by the renderer which is tied to realization of
+        the widget.</para></footnote>, and are realized for one single GL
+        Context at a time (or unrealized and independent). Once a GL resource
+        is not used anymore you can call gthree_resource_unrealize_for() to
+        unrealize all the resources for the content. Additionally, whenever
+        some object (like a scene) uses a resource it holds an extra refcount
+        (the use count), and when the use count goes to zero the resource will
```
Comment (alexlarsson) [inline docs/resources.xml]: Hmm, this is not actually true anymore. I changed resource_finalize() to always free the object, keeping only the GL state (e.g. the resource identifier) around for freeing later. I.e. there is no use count anymore and the resource objects don't (completely) hang around anymore.

File: docs/resources.xml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        e.g. rendering), this to avoid leaking GPU resources for unused
+        things.
+      </para>
+
+      <para>
+        Ownership wise this is a done with a reference cycle, where the GL
+        context object holds a ref to all resources (to avoid them being
+        finalized before unrealization). This cycle must be broken manually by
+        calling one of the gthree_resources_unrealize_* calls with the GL
+        context active (which is done automatically by the #GthreeArea widget).
```
Comment (alexlarsson) [inline docs/resources.xml]: So, this too should go per the above.

File: docs/gthree-docs.xml (hunk 1 above)
Comment (ebassi) [reply docs/gthree-docs.xml]: The first chapter is the basic types; the specializations are under their own section.

I copied the same structure as GTK, but it's just a first stab: I don't have any hard preference.

File: docs/resources.xml (hunk 2 above)
Comment (ebassi) [reply docs/resources.xml]: I literally copied the `resources.txt` file, but it may have changed between the time I copied and the time I rebased the PR.

Comment (alexlarsson) [reply docs/resources.xml]: I just update the code without changing the file. Lemme update the docs.

**Pull request merged**